### PR TITLE
Add ability to create a Registration

### PR DESCRIPTION
**Pull request by danieldpence**
@dtbateman With this change, I've added a `Registration` model, the corresponding `RegistrationsController`, as well as renamed a few things and a few architectural changes.

Next, we'll replace the static HTML you build using either Rails' built-in form builder or maybe bring in [simple_form](https://github.com/plataformatec/simple_form).

### CODE REVIEW

**Comment by danieldpence** [comment]
@dtbateman Alright man, I've added some more code here that will give you the ability to fully create a new registration, persist it to the database, and then see a confirmation page. I'll add some server-side model validations and a test or two next.

When you get a chance, cruise through the files I've added/changed here, ask any questions you might have, and then we'll merge this branch into the Master branch.

File: app/views/registrations/__new.html.erb [hunk 1]
```diff
@@ -1,18 +1,6 @@
-<!DOCTYPE html>
```
Comment (danieldpence) [inline app/views/registrations/__new.html.erb:1]: I left this file that you created in the app for reference, but renamed it so it wouldn't get in the way.

File: config/initializers/simple_form.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# Use this setup block to configure all options available in SimpleForm.
+SimpleForm.setup do |config|
```
Comment (danieldpence) [inline config/initializers/simple_form.rb:2]: This is an initializer file for the `SimpleForm` library I'm using to help build our forms. Don't be too concerned about it at this point.

File: config/locales/simple_form.en.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      text: 'required'
+      mark: '*'
+      # You can uncomment the line below if you need to overwrite the whole required html.
+      # When using html, text and mark won't be used.
+      # html: '<abbr title="required">*</abbr>'
+    error_notification:
+      default_message: "Please review the problems below:"
+    labels:
+      defaults:
+        first_name: "First Name"
```
Comment (danieldpence) [inline config/locales/simple_form.en.yml:15]: If you want to easily change the form labels generated by `SimpleForm`, this is where you'll need to do that. Each key here corresponds to the model attribute of the form field.

File: config/initializers/simple_form.rb (hunk 2 above)
Comment (dtbateman) [reply config/initializers/simple_form.rb:2]: so I've read about SimpleForm, but it seems that people have mixed feelings about it- what would you say are its advantages? 

Is it okay to go with RoR native form builder?

Comment (danieldpence) [reply config/initializers/simple_form.rb:2]: It's perfectly fine to start with the Rails native form builder and helpers, that's fine. Simple form mostly follows the same API, but it makes it real quick and easy to build forms. In terms of benefits, you configure it however you want and it automatically generates the html around each input field based on your settings. Also, it automatically handles error messages and makes things really easy to configure. We use it for our forms at work and have seen great success/flexibility.

File: app/models/registration.rb [hunk 4]
```diff
@@ -1,2 +1,7 @@
 class Registration < ApplicationRecord
+  validates :first_name, presence: true
```
Comment (danieldpence) [inline app/models/registration.rb:2]: These validations are super basic, essentially just requiring a value be present for the fields specified. The only one that's a little different is `:gender`, which is requiring that value submitted be included in the given array (male or female).

File: app/assets/stylesheets/registration.scss [hunk 5]
```diff
@@ -127,22 +127,19 @@ font-weight: bold;
...
 }
 
 @media screen and (min-width: 480px) {
-
   form {
     max-width: 480px;
   }
+}
 
+.field_with_errors .error {
```
Comment (danieldpence) [inline app/assets/stylesheets/registration.scss:143]: `.field_with_errors` is the default class name added to an input field wrapper containing an error by SimpleForm. This is configurable in the `simple_form.rb` initializer, [L9](https://github.com/dtbateman/fs_roster/pull/1/files/8e7156bcae7b3dbbeb45c04b11656b97cf9b5ee7#diff-754d908a36f8f0af1316fe4b5ecd2c9bR9).

**Pull request closed without merging**